spec/features/note_moving_spec.rb

```diff
@@ -0,0 +1,62 @@
+require "spec_helper"
+
+describe "moving a note", js: true do
+
+  subject { page }
+
+  before do
+    login_to_project_as_user
+
+    # Tree:
+    #
+    # - node_0
+    #   - node_2
+    #     - node_5
+    #   - node_3
+    # - node_1
+    #   - node_4
+    @node_0 = create_node(label: "Node 0")
+    @node_1 = create_node(label: "Node 1")
+    @node_2 = create_node(label: "Node 2", parent: @node_0)
+    @node_3 = create_node(label: "Node 3", parent: @node_0)
+    @node_4 = create_node(label: "Node 4", parent: @node_1)
+    @node_5 = create_node(label: "Node 4", parent: @node_2)
+
+    visit node_note_path(@node_5, current_note)
+    click_move_note
+  end
+
+  let(:current_note) { @note = create(:note, node: @node_5) }
+
+  describe "moving a note to a different node" do
+    before do
+      within('#modal_move_note') do
+        click_link @node_1.label
+        click_submit
+      end
+    end
+
+    it "should update the note's node_id" do
+      expect(current_note.reload.node_id).to eq(@node_1.id)
+    end
+
+    it "should redirect to note show path" do
+      expect(current_path).to eq(node_note_path(@node_1, current_note))
+    end
+  end
+
+
+
+  def create_node(attrs={})
+    create(:node, attrs)
```

Comment:
I don't really see what advantage this method provides over just calling `create(:node, ...)` directly?